src/audio/rtnr/rtnr.c

```diff
@@ -0,0 +1,632 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Realtek Semiconductor Corp. All rights reserved.
+ *
+ * Author: Ming Jen Tai <mingjen_tai@realtek.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/rtnr/rtnr.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <sof/audio/rtnr/rtklib/include/RTK_MA_API.h>
+
+#define MicNum 2
+#define SpkNum  2
+
+#define RTNR_BLK_LENGTH			4 /* Must be power of 2 */
+#define RTNR_BLK_LENGTH_MASK	(RTNR_BLK_LENGTH - 1)
+#define RTNR_MAX_SOURCES		1 /* Microphone stream */
+
+static const struct comp_driver comp_rtnr;
+
+/** \brief RTNR processing functions map item. */
+struct rtnr_func_map {
+	enum sof_ipc_frame fmt; /**< source frame format */
+	rtnr_func func; /**< processing function */
+};
+
+/* UUID 5c7ca334-e15d-11eb-ba80-0242ac130004 */
+DECLARE_SOF_RT_UUID("rtnr", rtnr_uuid, 0x5c7ca334, 0xe15d, 0x11eb, 0xba, 0x80,
+		    0x02, 0x42, 0xac, 0x13, 0x00, 0x04);
+
+DECLARE_TR_CTX(rtnr_tr, SOF_UUID(rtnr_uuid), LOG_LEVEL_INFO);
+
+/* Generic processing */
+
+void myPrintf(int a, int b, int c, int d, int e)
+{
```

Comment:
I think we already have a hexdump API upstream, can you check. Btw function names should use snake_case and not CamelCase.

Reply:
@lgirdwood The proprietary library is built on another system without sof infrastructure. So we created a wrapper function that can be called from the library. This function is renamed to meet the standard. Thanks.